BO_training/bo/optimizer_mixed.py

```diff
@@ -154,8 +154,14 @@ def restart(self):
                 )
                 X_init_cat.append(cat_sample)
             X_init_cat = np.array(X_init_cat)
-            self.X_init = np.hstack((X_init_cat, X_init_cont))
-
+            #DONE: change stack manar into insert according to dim of cat
+            #self.X_init = np.hstack((X_init_cat, X_init_cont))
+            col = 0
+            for i in self.d_cat:
```

Comment:
Use `for col, i in enumerate(self.d_cat):` and remove `col = 0` and `col += 1` lines. Simpler, shorter, and more explanatory.